pkg/liqo-controller-manager/offloadingStatus-controller/offloadingstatus_controller.go

```diff
@@ -0,0 +1,153 @@
+type OffloadingStatusReconciler struct {
+	client.Client
+	Scheme      *runtime.Scheme
+	RequeueTime time.Duration
+}
+
+// Controller Ownership:
+// --> NamespaceOffloading.Status.RemoteConditions
+// --> NamespaceOffloading.Status.OffloadingPhase
+
+func setNamespaceOffloadingStatus(noff *offv1alpha1.NamespaceOffloading) {
+	ready := 0
+	notReady := 0
+
+	for i := range noff.Status.RemoteNamespacesConditions {
+		condition := liqoutils.FindRemoteNamespaceCondition(noff.Status.RemoteNamespacesConditions[i], offv1alpha1.NamespaceReady)
+		if condition == nil {
+			continue
+		}
+		if condition.Status == corev1.ConditionTrue {
+			ready++
+		} else {
+			notReady++
+		}
+	}
+
+	switch {
+	case !noff.DeletionTimestamp.IsZero():
+		noff.Status.OffloadingPhase = offv1alpha1.TerminatingOffloadingPhaseType
+		if ready+notReady == 0 {
+			// The NamespaceOffloading is deleted only when there are no more remoteNamespaceCondition and
+			// the deletion timestamp is set.
+			for key := range noff.Status.RemoteNamespacesConditions {
+				delete(noff.Status.RemoteNamespacesConditions, key)
+			}
+			klog.Infof("NamespaceOffloading, in the namespace '%s', ready to be deleted", noff.Namespace)
+		}
+	case ready+notReady == 0:
+		noff.Status.OffloadingPhase = offv1alpha1.NoClusterSelectedOffloadingPhaseType
+	case ready == 0:
+		noff.Status.OffloadingPhase = offv1alpha1.AllFailedOffloadingPhaseType
+	case notReady == 0:
+		noff.Status.OffloadingPhase = offv1alpha1.ReadyOffloadingPhaseType
+	default:
+		noff.Status.OffloadingPhase = offv1alpha1.SomeFailedOffloadingPhaseType
+	}
+
+	klog.Infof("The OffloadingStatus for NamespaceOffloading in the namespace '%s' is set to '%s'",
+		noff.Namespace, noff.Status.OffloadingPhase)
+}
+
+// Reconcile sets the NamespaceOffloading Status checking the actual status of all remote Namespace.
+// This reconcile is performed every RequeueTime.
+func (r *OffloadingStatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	namespaceOffloading := &offv1alpha1.NamespaceOffloading{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: req.Namespace,
+		Name:      liqoconst.DefaultNamespaceOffloadingName,
+	}, namespaceOffloading); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("There is no NamespaceOffloading resource in Namespace '%s'", req.Namespace)
+			return ctrl.Result{}, nil
+		}
+		klog.Errorf("%s --> Unable to get namespaceOffloading for the namespace '%s'", err, req.Namespace)
+		return ctrl.Result{}, err
+	}
+
+	// Get all NamespaceMaps in the cluster
+	namespaceMapList := &mapsv1alpha1.NamespaceMapList{}
+	if err := r.List(ctx, namespaceMapList); err != nil {
+		klog.Errorf("%s -> unable to get NamespaceMaps", err)
+		return ctrl.Result{}, err
+	}
+
+	if len(namespaceMapList.Items) == 0 {
+		klog.Info("No NamespaceMaps in the cluster at the moment")
+		return ctrl.Result{RequeueAfter: r.RequeueTime}, nil
+	}
+
+	patch := namespaceOffloading.DeepCopy()
```

Comment:
```suggestion
	original := namespaceOffloading.DeepCopy()
```

Nit. Just a bit more clear

Reply:
thank you